drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -299,6 +390,30 @@ local function assign_child_profile(device, child_ep)
   return profile or "switch-binary"
 end
 
+local function do_configure(driver, device)
+  local energy_eps = embedded_cluster_utils.get_endpoints(device, clusters.ElectricalEnergyMeasurement.ID)
+  local power_eps = embedded_cluster_utils.get_endpoints(device, clusters.ElectricalPowerMeasurement.ID)
+  local valve_eps = embedded_cluster_utils.get_endpoints(device, clusters.ValveConfigurationAndControl.ID)
+  local profile_name = nil
+  if #energy_eps > 0 and #power_eps > 0 then
+    profile_name = "plug-power-energy-powerConsumption"
```

Comment:
This handling will eventually need to be added to `initialize_switch()` so that on/off plugs with energy metering can be handled as well. However, since we do not know of any devices that exist that do this yet, this can be done after MVP.

Reply:
That's exactly what Nick and I discussed when adding this :+1: with the plans to add this logic as the market advanced.